src/main/java/pacApp/pacSecurity/JwtAuthenticationTokenFilter.java

```diff
@@ -9,28 +11,47 @@
 
 import javax.servlet.FilterChain;
 import javax.servlet.ServletException;
+import javax.servlet.http.Cookie;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.io.IOException;
 
 @Component
 public class JwtAuthenticationTokenFilter extends OncePerRequestFilter {
 
+    private static final Logger log = LoggerFactory.getLogger(JwtAuthenticationTokenFilter.class);
+
     @Value("${jwt.header}")
     private String tokenHeader;
 
     @Override
     protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
-        final String requestHeader = request.getHeader(this.tokenHeader);
+        this.authenticateToken(request, response);
+        filterChain.doFilter(request, response);
+    }
 
-        if(requestHeader != null && requestHeader.startsWith("Bearer ")){
-            String authToken = requestHeader.substring(7);
-            JwtAuthentication authentication = new JwtAuthentication(authToken);
+    protected void authenticateToken(HttpServletRequest request, HttpServletResponse response) {
+        String authToken = null;
+        Cookie[] cookies = request.getCookies();
 
-            this.updateSecurityContextAuthentication(authentication);
+        if (cookies == null) {
+            log.info("cookies is null");
+            return;
         }
 
-        filterChain.doFilter(request, response);
+        for (Cookie cookie : cookies) {
+            log.info(cookie.getName());
+            if (cookie.getName().equals("token")) {
+                authToken = cookie.getValue();
```

Comment:
this is the autehntication? why grep the value from the cookie and create a new token with it?

Reply:
No, this part of the JwtAuthenticationFilter only searches the token in the incoming request. If the token was found, the token is saved afterwards in the SecurityContext.

The authentication is ultimately handled by an instance of the JwtAuthenticationProvider that is registered in the WebSecurityContext.

The implementation of authentication via JWT has not changed with the code introduced by this pull request.

Reply:
Okay fine with me